modules/tinymce/src/plugins/searchreplace/test/ts/browser/SearchReplaceKeyboardNavigationTest.ts

```diff
@@ -1,121 +1,110 @@
-        tinyApis.sSetContent('<p>fish fish fish</p>'),
-        sOpenDialog(tinyUi),
-        sAssertFocused('Find input', '.tox-textfield[placeholder="Find"]'),
-        sPressTab,
-        sAssertFocused('Replace with input', '.tox-textfield[placeholder="Replace with"]'),
-        sPressTab,
-        sAssertFocused('Placeholder menu button', '.tox-tbtn--select[title="Preferences"]'),
-        sPressTab,
-        sAssertFocused('Find button', '.tox-button[title="Find"]'),
-        Chain.asStep(body, [
-          UiFinder.cFindIn('input.tox-textfield[placeholder="Find"]'),
-          UiControls.cSetValue('fish')
-        ]),
-        sPressEnter,
-        sPressTab,
-        sAssertFocused('Find button', '.tox-button[title="Replace"]'),
-        sPressTab,
-        sAssertFocused('Find button', '.tox-button[title="Replace all"]'),
-        sPressEnter,
-        sAssertFocused('Find input', '.tox-textfield[placeholder="Find"]'),
-        sPressEsc
-      ]),
-      Log.stepsAsStep('TINY-4014', 'Find and replace: Dialog keyboard focus is returned to find input after displaying an alert', [
-        tinyApis.sSetContent('<p>fish fish fish</p>'),
-        sOpenDialog(tinyUi),
-        sAssertFocused('Find input', '.tox-textfield[placeholder="Find"]'),
-        Chain.asStep(body, [
-          UiFinder.cFindIn('input.tox-textfield[placeholder="Find"]'),
-          UiControls.cSetValue('notfound')
-        ]),
-        sPressEnter,
-        sAssertFocused('Alert dialog OK button', '.tox-alert-dialog .tox-button[title="OK"]'),
-        sPressEnter,
-        sAssertFocused('Find input', '.tox-textfield[placeholder="Find"]'),
-        sPressEsc
-      ])
-    ], onSuccess, onFailure);
-  }, {
-    theme: 'silver',
+describe('browser.tinymce.plugins.searchreplace.SearchReplaceKeyboardNavigationTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
     plugins: 'searchreplace',
     toolbar: 'searchreplace',
-    base_url: '/project/tinymce/js/tinymce',
     menubar: 'file edit',
     menu: {
       file: { title: 'File', items: 'newdocument' },
       edit: { title: 'Edit', items: 'selectall searchreplace' }
-    }
-  }, success, failure);
+    },
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ], true);
+
+  const doc = SugarDocument.getDocument();
+
+  const pressTab = () => Keyboard.activeKeydown(doc, Keys.tab());
+  const pressEsc = () => Keyboard.activeKeydown(doc, Keys.escape());
+  const pressDown = () => Keyboard.activeKeydown(doc, Keys.down());
+  const pressRight = () => Keyboard.activeKeydown(doc, Keys.right());
+  const pressEnter = () => Keyboard.activeKeydown(doc, Keys.enter());
+
+  const focusToolbar = (editor: Editor) => {
+    const args = Tools.extend({
+      ctrlKey: false,
+      altKey: false,
+      shiftKey: false,
+      metaKey: false
+    }, { altKey: true, keyCode: 120 });
+    editor.fire('keydown', args);
+  };
```

Comment:
```suggestion
  const focusToolbar = (editor: Editor) =>
    TinyContentActions.keydown(editor, 120, { altKey: true });
```
This is just a more accurate action and I'd like to avoid any `editor.fire()` calls if we can help it. Not really in scope though, so your call.